odoo/sql_db.py

```diff
@@ -278,49 +237,48 @@ class Cursor(BaseCursor):
     """
     IN_MAX = 1000   # decent limit on size of IN queries - guideline = Oracle limit
 
-    def __init__(self, pool, dbname, dsn, serialized=True):
+    def __init__(self, pool, dbname, dsn, **kwargs):
         super().__init__()
-
+        if 'serialized' in kwargs:
+            warnings.warn("Since 16.0, 'serialized' parameter is not used anymore.", DeprecationWarning, 2)
+        assert not (kwargs.keys() - {'serialized'})
         self.sql_from_log = {}
         self.sql_into_log = {}
 
         # default log level determined at cursor creation, could be
         # overridden later for debugging purposes
-        self.sql_log = _logger.isEnabledFor(logging.DEBUG)
-
         self.sql_log_count = 0
 
         # avoid the call of close() (by __del__) if an exception
-        # is raised by any of the following initialisations
+        # is raised by any of the following initializations
         self._closed = True
 
         self.__pool = pool
         self.dbname = dbname
-        # Whether to enable snapshot isolation level for this cursor.
-        # see also the docstring of Cursor.
-        self._serialized = serialized
 
         self._cnx = pool.borrow(dsn)
         self._obj = self._cnx.cursor()
-        if self.sql_log:
+        if _logger.isEnabledFor(logging.DEBUG):
             self.__caller = frame_codeinfo(currentframe(), 2)
         else:
             self.__caller = False
-        self._closed = False   # real initialisation value
-        self.autocommit(False)
-
-        self._default_log_exceptions = True
+        self._closed = False   # real initialization value
+        # See the docstring of this class.
+        self.connection.set_isolation_level(ISOLATION_LEVEL_REPEATABLE_READ)
```

Comment:
Do we have a test that verifies the isolation level of new cursors? I could not find any with a quick search. If we're touching that core piece of logic, it seems an important (and easy) thing to cover, no?

Reply:
Agree, I will add it.